gobblin-utility/src/main/java/org/apache/gobblin/util/filters/RegexPathFilter.java

```diff
@@ -34,6 +35,11 @@
 
   private final Pattern regex;
   private final boolean include;
+  public static final String REGEX = "path.filter.regex";
```

Comment:
So, is this path filter only used to filter source datasets?
I can see that we specify the class using gobblin.dataset.path.filter.class . It could make sense to have the same "gobblin.dataset" prefix for this setting for consistency.

Also, it could be better to keep those config settings together, so that they would be easily discovered.

Usually, they land up in ConfigKeys class, but I can see that bunch of keys are defined in DatasetUtils. You can try to figure out if there is any reason to keep them there.

Reply:
imo, this and other path filters can be used to filter any path. They are not tied up with filtering only gobblin datasets.
In past, there have been discussion about where to keep configs, a) in ConfigurationKeys or somewhere else along with other configs or b) in the class which actually use them.
For now, I will move it to DatasetUtils

Reply:
Ah, I cannot move it to DatasetUtils because that would add a circular dependency between gobblin-utility and gobblin-data-management